src/App.jsx

```diff
@@ -2,48 +2,45 @@
 import { useEffect, useState } from 'react';
 import { AddItem, Home, Layout, List } from './views';
 import { Routes, Route, useNavigate } from 'react-router-dom';
+import { getItemData, streamListItems } from './api';
 import { useStateWithStorage } from './utils';
 import { generateToken } from '@the-collab-lab/shopping-list-utils';
 
 export function App() {
 	const navigate = useNavigate();
-	const [displayName, setDisplayName] = useState('');
+	const [data, setData] = useState([]);
+
 	const [listToken, setListToken] = useStateWithStorage(
 		null,
 		'tcl-shopping-list-token',
 	);
 
-	useEffect(() => {
-		if (listToken) navigate('/list');
-	}, []);
-
 	function handleClick() {
 		if (listToken) return;
 		const token = generateToken();
 		setListToken(token);
 	}
 
-	const handleInputChange = (evt) => {
-		setDisplayName(evt.target.value);
-	};
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+
+			return streamListItems(listToken, (snapshot) => {
+				const nextData = getItemData(snapshot);
+
+				setData(nextData);
+			});
```

Comment:
Why do you choose to load the list data in the `App` component instead of the `List` component. From what I understand you only need the list data in the list component so this logic can be loaded there. It's good to lift state up when data needs to be shared between components but I think in this case it's not necessary.

Reply:
Hi Golfo,

This is part of the boilerplate code we have in this project and I did not want to remove that. :)